src/system/startup.c

```diff
@@ -21,31 +21,33 @@
 extern void rtos_initialize();
 extern void vfs_initialize();
 extern void system_daemon_initialize();
-// extern void graphical_context_daemon_initialize(void);
-extern void display_initialize(void);
 extern void rtos_sched_start();
 extern void vdml_initialize();
 extern void invoke_install_hot_table();
 
-// XXX: pros_init happens inside __libc_init_array, and before any global
-// C++ constructors are invoked. This is accomplished by instructing
-// GCC to include this function in the __init_array. The 101 argument
-// gives the compiler instructions on the priority of the constructor,
-// from 0-~65k. The first 0-100 priorities are reserved for language
-// implementation.
-__attribute__((constructor(101))) static void pros_init(void) {
+// XXX: pros_pre_init and pros_init happen inside __libc_init_array, and
+// before any globalC++ constructors are invoked. This is accomplished by
+// instructing GCC to include this function in the __init_array. The macros
+// defined in pros/apix.h give the compiler instructions on the priority of the
+// constructor, from 0-~65k. The first 0-100 priorities are reserved for
+// language implementation.
+
+__attribute__((constructor(PROS_KERNEL_PRE_INIT))) static void pros_pre_init(void) {
 	rtos_initialize();
 
 	vfs_initialize();
 
 	vdml_initialize();
+}
 
-	display_initialize();
+// HACK: External templates can hook into initialization process by setting
+// their priority between PROS_KERNEL_PRE_INIT and PROS_KERNEL_INIT. For
+// exmaple the priority of display initialization is 119. All other initialize
+// functions should be called before pros_init function. For an exmaple of
+// what could happen if this is not the case, see
+// https://github.com/purduesigbots/pros/pull/144/#issuecomment-496901942
```

Comment:
I don't think this comment it right, or at least the reasoning behind this specific case. There was no user code being run between the different parts of initialization (as the scheduler wasn't started yet), so the only way the ordering would matter is if it's relying on the order of tasks being started.